MapboxNavigation/NavigationViewController.swift

```diff
@@ -395,8 +406,10 @@ open class NavigationViewController: UIViewController {
     
     open override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
-        
-        UIApplication.shared.isIdleTimerDisabled = true
+
+        if delegate?.navigationViewController?(self, shouldSetIsIdleTimerDisabledTo: true) ?? true {
```

Comment:
I hadn’t seen the discussion up to now, but I kind of agree with https://github.com/mapbox/mapbox-navigation-ios/pull/1591#issuecomment-411525350 that it would be better to expose this option as a Boolean flag rather than a delegate method. A NavigationViewController instance only ever calls this method twice at predictable times in its life cycle. Battery management is more nuanced than the idle timer; I suspect the application would know upfront whether the SDK should handle the idle timer automatically or not.

Do you have a strong preference for either approach, @macdrevx? Let us know if you’d like us to take care of going back to the Boolean flag for you. Sorry for the back and forth. 😅

Reply:
@1ec5 @bsudekum I'm happy to do it either way (and happy to change it back if that's what the team prefers). Just let me know once you decide and I can make the change if needed.

Reply:
@macdrevx @1ec5 I'm okay with a property, thanks for your flexibility @macdrevx.

Reply:
Sounds good. Changes forthcoming…